### PR TITLE
ANN: Cancel cargo check process on any PSI change

### PR DESCRIPTION
**Pull request by mchernyavsky**
Depends on https://github.com/intellij-rust/intellij-rust/pull/3466.

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/cargoCheck/RsCargoCheckUtils.kt [hunk 1]
```diff
@@ -172,6 +175,20 @@ fun AnnotationHolder.createAnnotationsForFile(file: RsFile, annotationResult: Rs
     }
 }
 
+fun Disposable.createChildDisposedOnAnyPsiChange(messageBus: MessageBus): Disposable {
+    val child = Disposable {}
+    Disposer.register(this, child)
+    messageBus.connect().subscribe(
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/cargoCheck/RsCargoCheckUtils.kt]: This is a memory leak: you register a listener and never unregister it. Looks like you should write `messageBus.connect(child)`

Comment (mchernyavsky) [reply src/main/kotlin/org/rust/ide/annotator/cargoCheck/RsCargoCheckUtils.kt]: Fixed.

File: src/main/kotlin/org/rust/ide/annotator/cargoCheck/RsCargoCheckAnnotatorPass.kt [hunk 2]
```diff
@@ -49,40 +50,35 @@ class RsCargoCheckAnnotatorPass(
         if (cargoPackage?.origin != PackageOrigin.WORKSPACE) return
 
         val project = file.project
+        disposable = ModuleUtil.findModuleForFile(file) ?: project
+        disposable = disposable.createChildDisposedOnAnyPsiChange(project.messageBus)
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/cargoCheck/RsCargoCheckAnnotatorPass.kt]: Why disposable is reassigned?

**Pull request merged**